src/evaluate.jl

```diff
@@ -112,8 +112,11 @@ evaluate(p::Taylor1{T}, x::Array{S}) where {T<:Number, S<:Number} =
 (p::Taylor1)() = evaluate(p)
 
 #function-like behavior for Vector{Taylor1}
-(p::AbstractArray{Taylor1{T}})(x) where {T<:Number} = evaluate.(p, x)
-(p::AbstractArray{Taylor1{T}})() where {T<:Number} = evaluate.(p)
```

Comment:
Maybe you need `AbstractArray{<:Taylor1{T}}`?

Reply:
I tried it a couple of times; tests (for Julia v1.0) did not pass. So I reverted it to an older version.

Reply:
Personally I wouldn't worry about making things work on Julia 1.0. We should just bump the minimum Julia version.

Reply:
You are probably right; Julia 1.5 is around the corner. Which version of Julia would you suggest to use?

Reply:
Let's say 1.4 I guess.

Reply:
Julia 1.3 is the version where it was allowed to define methods for abstract types. I'm pushing a commit with this as the minimal Julia version, and also the changes in the function-like evaluation.